### PR TITLE
untracked: Make  and  parameters to the Factbase API compulsory so th…

### PR DESCRIPTION
**Pull request by oliverbock**
…at when an upload stops we have enough information to work out how to fix it.

### CODE REVIEW

File: R/Factbase.R [hunk 1]
```diff
@@ -342,6 +345,20 @@ value_type_for_vector <- function(v, column_name) {
         stop(paste('Cannot work out which data type to use for column', column_name, 'containing a', typeof(v), 'vector.  Only Date, POSIXt, text or real are accepted'))
 }
 
+ensureDefinitionHyperlinkOwnerSupplied <- function(definition, hyperlink, owner) {
+    if (is.null(definition))
+        stop("argument \"definition\" is missing")
+    if (!is.character(definition) || length(definition) != 1)
+        stop("argument \"definition\" must be a character vector of length 1, or null")
```
Comment (jrwishart) [inline R/Factbase.R]: Stating `"definition"` can be be `NULL` contradicts the lines above. 

A related aside, `is.character(NULL)` returns `FALSE` so can be removed unless you wanted visibility of a missing arg?

File: R/Factbase.R [hunk 2]
```diff
@@ -342,6 +345,20 @@ value_type_for_vector <- function(v, column_name) {
...
+    if (is.null(hyperlink))
+        stop("argument \"hyperlink\" is missing")
+    if (!is.character(hyperlink) || length(hyperlink) != 1)
+        stop("argument \"hyperlink\" must be a character vector of length 1")
+    if (is.null(owner))
+        stop("argument \"owner\" is missing")
+    if (!is.character(owner) || length(owner) != 1)
+        stop("argument \"owner\" must be a character vector of length 1")
+}
```
Comment (jrwishart) [inline R/Factbase.R:361]: FYI: I think it is easier to consolidate these conditions into the simpler `stopifnot` function when the `stop` message isn't too long and the condition isn't too long.

```
stopifnot("argument \"owner\" must be a character vector of length 1" = is.character(owner) && length(owner) == 1L)
```

**Review by jrwishart — APPROVED** [review]
Approving. Left some comments that might be worth addressing but shouldn't block merging.

File: R/Factbase.R (hunk 1 above)
Comment (oliverbock) [reply R/Factbase.R]: Good spot.

File: R/Factbase.R (hunk 2 above)
Comment (oliverbock) [reply R/Factbase.R:361]: I tried this out and I found the code harder to read, so I'm leaving it as is.

**Pull request merged**